### ISSUE TITLE
Improve transparency with Pytorch and Pytorch lightning

### ISSUE THREAD

**Issue by SalamanderXing** [ilex-paraguariensis/yerbamate#34]
Right now, we have to manually add all of the PyTorch optimizers from the parameters and the same thing for scheduler and EarlyStopping.

It would be nice if the configuration file would be a mirror to these objects in their libraries (PyTorch and lightning). This way people wouldn't have to learn to use our optimizers but could just look into these libraries' documentation.

**Comment by SalamanderXing** [comment]
Now the optimizer section of config is the same as the return for lightning configure optimizer, that is:

```
"configure_optimizers":{
	  "optimizer": {
		  "type":"Adam",
		  "lr": 0.0004,
		  "betas": [0.5, 0.999]
	  },
	  "lr_scheduler": {
		"monitor": "val_loss",
		"scheduler": {
		  "type":"ReduceLROnPlateau",
		  "mode": "min",
		  "factor": 0.5,
		  "threshold": 1e-8,
		  "threshold_mode": "rel",
		  "patience": 0,
		  "verbose": true
		}
	  }
   },
```
Notice that objects "scheduler" and "optimizers" will directly initialize an object of class "type" with the same in PyTorch. Also, the arguments will be exactly the same as in the official documentation.

This also implies that now mate supports **all** of the optimizers and schedulers supported by PyTorch.

Also now EarlyStopping is defined by its own little (optional) section:
```
  "early_stopping": {
    "monitor": "val_loss",
	"patience": 5,
	"verbose": true,
	"mode": "min",
	"strict":false
  },
```
Again, this is a mirror to the official EarlyStopping by Lightning (the parameters will be called exactly the same).

Example of a full valid config:
```
{
  "configure_optimizers":{
	  "optimizer": {
		  "type":"Adam",
		  "lr": 0.0004,
		  "betas": [0.5, 0.999]
	  },
	  "lr_scheduler": {
		"monitor": "val_loss",
		"scheduler": {
		  "type":"ReduceLROnPlateau",
		  "mode": "min",
		  "factor": 0.5,
		  "threshold": 1e-8,
		  "threshold_mode": "rel",
		  "patience": 0,
		  "verbose": true
		}
	  }
   },
  "early_stopping": {
    "monitor": "val_loss",
	"patience": 5,
	"verbose": true,
	"mode": "min",
	"strict":false
  },
  "out_len": 15,
  "crop_size": [200, 200],
  "in_features": 188,
  "cuda": true,
  "train_batch_size": 16,
  "test_batch_size": 16,
  "max_epochs": 100,
  "early_stopping_patience": 5,
  "data_location": "/mnt/foci_detection.hdf5",
  "data_loader": "foci",
  "normalization": "none"
}
```
